src/main/java/io/fabric8/vertx/maven/plugin/mojos/SetupMojo.java

```diff
@@ -0,0 +1,160 @@
+/*
+ *   Copyright 2016 Kamesh Sampath
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.fabric8.vertx.maven.plugin.mojos;
+
+import io.fabric8.vertx.maven.plugin.utils.MojoUtils;
+import org.apache.maven.model.*;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.dependency;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.plugin;
+
+/**
+ * This Goal helps in setting up Vert.x maven project with vertx-maven-plugin, with sane defaults
+ */
+@Mojo(name = "setup")
+public class SetupMojo extends AbstractMojo {
+
+    final String PLUGIN_GROUPID = "io.fabric8";
+    final String PLUGIN_ARTIFACTID = "vertx-maven-plugin";
+    final String VERTX_MAVEN_PLUGIN_VERSION_PROPERTY = "vertx-maven-plugin-version";
+    final MojoUtils mojoUtils = new MojoUtils();
+
+    /**
+     * The Maven project which will define and confiure the vertx-maven-plugin
+     */
+    @Parameter(defaultValue = "${project}", readonly = true)
+    protected MavenProject project;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        Model model = project.getModel();
+        File pomFile = project.getFile();
+
+        Optional<Plugin> vmPlugin = mojoUtils.hasPlugin(project, "io.fabric8:vertx-maven-plugin");
```

Comment:
even i dont know either, we can have a issue opened for it to track and if we see some light we can update the plugin